pkg/virt-launcher/virtwrap/network/common.go

```diff
@@ -422,70 +455,13 @@ func buildTapDeviceMaker(tapName string, queueNumber uint32, virtLauncherPID int
 	// #nosec No risk for attacket injection. createTapDeviceArgs includes predefined strings
 	cmd := exec.Command("virt-chroot", createTapDeviceArgs...)
 
-	return &tapDeviceMaker{
-		tapMakingCmd:    cmd,
-		tapName:         tapName,
-		queueNumber:     queueNumber,
-		virtLauncherPID: virtLauncherPID,
-	}, nil
-}
-
-func (tmc *tapDeviceMaker) makeTapDevice() error {
-	if isSELinuxEnabled() {
-		defer func() {
-			_ = resetVirtHandlerSELinuxContext()
-		}()
-
-		if err := setVirtLauncherSELinuxContext(tmc.virtLauncherPID); err != nil {
-			return err
-		}
-	}
-
-	preventFDLeakOntoChild()
-	if err := tmc.tapMakingCmd.Run(); err != nil {
-		return fmt.Errorf("failed to create tap device %s: %v", tmc.tapName, err)
-	}
-	return nil
-}
-
-func isSELinuxEnabled() bool {
-	_, selinuxEnabled, err := kvselinux.NewSELinux()
-	return err == nil && selinuxEnabled
-}
-
-func setVirtLauncherSELinuxContext(virtLauncherPID int) error {
-	virtLauncherSELinuxLabel, err := getProcessCurrentSELinuxLabel(virtLauncherPID)
-	if err != nil {
-		return fmt.Errorf("error reading virt-launcher %d selinux label. Reason: %v", virtLauncherPID, err)
-	}
-
-	runtime.LockOSThread()
-	if err := selinux.SetExecLabel(virtLauncherSELinuxLabel); err != nil {
-		return fmt.Errorf("failed to switch selinux context to %s. Reason: %v", virtLauncherSELinuxLabel, err)
-	}
-	return nil
-}
-
-func resetVirtHandlerSELinuxContext() error {
-	virtHandlerSELinuxLabel := "system_u:system_r:spc_t:s0"
-	err := selinux.SetExecLabel(virtHandlerSELinuxLabel)
-	runtime.UnlockOSThread()
-	return err
-}
-
-func getProcessCurrentSELinuxLabel(pid int) (string, error) {
-	launcherSELinuxLabel, err := selinux.FileLabel(fmt.Sprintf("/proc/%d/attr/current", pid))
-	if err != nil {
-		return "", fmt.Errorf("could not retrieve pid %d selinux label: %v", pid, err)
-	}
-	return launcherSELinuxLabel, nil
-}
-
-func preventFDLeakOntoChild() {
-	// we want to share the parent process std{in|out|err}
-	for fd := 3; fd < 256; fd++ {
-		syscall.CloseOnExec(fd)
+	tapDeviceMaker := &tapDeviceMaker{
+		tapName:     tapName,
+		queueNumber: queueNumber,
 	}
+	tapDeviceMaker.cmdToExecute = cmd
```

Comment:
Why did you move those two out of the struct creation and how this change is related to the commit?